omod/src/main/java/org/openmrs/module/spa/SpaActivator.java

```diff
@@ -14,16 +14,35 @@
 import org.openmrs.api.context.Context;
 import org.openmrs.module.BaseModuleActivator;
 
+import javax.servlet.ServletContext;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.openmrs.module.spa.SpaConstants.BUNDLED_FRONTEND_DIRECTORY;
+import static org.openmrs.module.spa.SpaConstants.GP_LOCAL_DIRECTORY;
+
 @Slf4j
 public class SpaActivator extends BaseModuleActivator {
 
 	// here so we can register the listener on load and de-register it when stopped
 	private GlobalPropertyListener spaDirectoryResolver = null;
 
+	public static ServletContext servletContext;
+
 	@Override
 	public void started() {
 		log.info("SPA module started");
-		spaDirectoryResolver = new SpaDirectoryResolver();
+	}
+
+	@Override
+	public void contextRefreshed() {
+		SpaDirectoryResolver spaDirectoryResolver = new SpaDirectoryResolver();
+		if (!Paths.get(SpaDirectoryResolver.getSpaDirectory(), "index.html").toFile().exists()) {
+			Path bundledFrontend = Paths.get(servletContext.getRealPath("/"), "WEB-INF", BUNDLED_FRONTEND_DIRECTORY);
+			if (bundledFrontend.resolve("index.html").toFile().exists()) {
+				Context.getAdministrationService().setGlobalProperty(GP_LOCAL_DIRECTORY, bundledFrontend.toAbsolutePath().toString());
+			}
+		}
 		Context.getAdministrationService().addGlobalPropertyListener(spaDirectoryResolver);
```

Comment:
This code should probably have a guard if it's not in `started()`:

```
if (spaDirectoryResolver != null) {
	Context.getAdministrationService().addGlobalPropertyListener(spaDirectoryResolver);
}
```

This is because `contextRefreshed()` is called before `started()` IIRC.